File: lib/java/opentoken/src/main/java/com/truveta/opentoken/tokens/TokenGenerationException.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokens;

/**
 * Exception thrown when token generation fails.
 * <p>
 * This exception is thrown when there are errors during the token generation
 * process, such as invalid input data, validation failures, or processing errors.
 * </p>
 */
public class TokenGenerationException extends Exception {

    /**
     * Constructs a new TokenGenerationException with the specified detail message.
     *
     * @param message the detail message explaining the reason for the exception
     */
    public TokenGenerationException(String message) {
        super(message);
    }

    /**
     * Constructs a new TokenGenerationException with the specified detail message and cause.
     *
     * @param message the detail message explaining the reason for the exception
     * @param cause the underlying cause of this exception
     */
    public TokenGenerationException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: lib/nodejs/opentoken/src/tokens/TokenGenerationException.ts
/**
 * Copyright (c) Truveta. All rights reserved.
 */

/**
 * Exception thrown when token generation fails.
 *
 * This exception is thrown when there are errors during the token generation
 * process, such as invalid input data, validation failures, or processing errors.
 */
export class TokenGenerationException extends Error {
  cause?: Error;

  /**
   * Constructs a new TokenGenerationException with the specified detail message.
   *
   * @param message - The detail message explaining the reason for the exception.
   * @param cause - The underlying cause of this exception (optional).
   */
  constructor(message: string, cause?: Error) {
    super(message);
    this.name = 'TokenGenerationException';
    
    // Maintain proper stack trace for where our error was thrown (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, TokenGenerationException);
    }

    // Store the cause if provided
    if (cause) {
      this.cause = cause;
    }
  }
}
